src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -37,35 +36,37 @@ public ShoppingCart add(ShoppingCart shoppingCart) {
     }
 
     @Override
-    public ShoppingCart getByUser(User user) {
+    public Optional<ShoppingCart> getByUser(User user) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            Query<ShoppingCart> query = session.createQuery("FROM ShoppingCart sc "
-                    + "LEFT JOIN FETCH sc.tickets t "
-                    + "LEFT JOIN FETCH t.movieSession ms "
-                    + "LEFT JOIN FETCH ms.movie "
-                    + "LEFT JOIN FETCH ms.cinemaHall "
-                    + "WHERE sc.user =:user", ShoppingCart.class);
-            query.setParameter("user", user);
-            return query.uniqueResult();
+            return session.createQuery("from ShoppingCart s "
```

Comment:
The query uses `left join fetch` to eagerly fetch associated entities. Ensure that this is the desired behavior, as it may impact performance by loading more data than necessary. Consider using `join fetch` if you don't need to include null associations.